parsing/g_proofs.ml4

```diff
@@ -19,6 +19,25 @@ let thm_token = G_vernac.thm_token
 
 let hint = Gram.entry_create "hint"
 
+let warn_deprecated_focus =
+  CWarnings.create ~name:"deprecated-focus" ~category:"deprecated"
+         (fun () ->
+           Pp.strbrk
+             "The Focus command is deprecated; use bullets or focusing brackets instead"
+         )
+
+let warn_deprecated_focus_n n =
+  CWarnings.create ~name:"deprecated-focus" ~category:"deprecated"
+         (fun () ->
+           Pp.strbrk ( "The Focus command is deprecated; use "
+                       ^ string_of_int n ^ ": { instead"
```

Comment:
Please don't use `Pp.strbrk` and `^` together; no point in concatenating strings if they are split right afterwards, just use `++`. Also, don't use `strbrk` unless you are sure the message can be split at any whitespace; here the message cannot be split between the colon and the curly bracket. Also I suggest using delimiters to clearly show to the user what is supposed to be written.

```
Pp.(str "The Focus command is deprecated;" ++ spc () ++ str "use '" ++ int n ++ ": {' instead")
```

Reply:
Thanks for the suggestion @silene! In fact, I just copied the warning code from somewhere else without really thinking about what `Pp.strbrk` was doing.